sklearn/metrics/_scorer.py

```diff
@@ -101,12 +101,23 @@ def __call__(self, estimator, *args, **kwargs):
         cache = {} if self._use_cache(estimator) else None
         cached_call = partial(_cached_call, cache)
 
+        all_scorer_failed = True
         for name, scorer in self._scorers.items():
-            if isinstance(scorer, _BaseScorer):
-                score = scorer._score(cached_call, estimator, *args, **kwargs)
+            try:
+                if isinstance(scorer, _BaseScorer):
+                    score = scorer._score(cached_call, estimator, *args, **kwargs)
+                else:
+                    score = scorer(estimator, *args, **kwargs)
+                scores[name] = score
+            except Exception as e:
+                scores[name] = e
+                last_exception = e
             else:
-                score = scorer(estimator, *args, **kwargs)
-            scores[name] = score
+                all_scorer_failed = False
+
+        if all_scorer_failed:
+            raise last_exception
```

Comment:
This is for passing the `test_cross_validate_failing_scorer` test in `test_validation.py`.